GitUI/CommandsDialogs/SettingsDialog/Pages/GitConfigSettingsPage.cs

```diff
@@ -250,6 +250,32 @@ private void txtDiffMergeToolPath_LostFocus(object sender, EventArgs e)
             }
         }
 
+        private void txtMergeToolPath_TextChanged(object sender, EventArgs e)
+        {
+            if (txtMergeToolPath.Focused && sender == txtMergeToolPath)
+            {
+                // we only want to suggest command for known tool types
+                var toolName = _NO_TRANSLATE_cboMergeTool.Text;
+                if (!string.IsNullOrWhiteSpace(toolName))
+                {
+                    SuggestMergeToolCommand();
+                }
+            }
+        }
+
+        private void txtDiffToolPath_TextChanged(object sender, EventArgs e)
+        {
+            if (txtDiffToolPath.Focused && sender == txtDiffToolPath)
+            {
+                // we only want to suggest command for known tool types
+                var toolName = _NO_TRANSLATE_cboDiffTool.Text;
+                if (!string.IsNullOrWhiteSpace(toolName))
```

Comment:
Still an issue, should be checked for difftools known by GE. If you add your own tool and call it like `kdiff3_updated` then the cmd is erased at all changes.

Reply:
Ok. I added verification for built-in tools. Please note that "Suggest" button doesn't care about such verification.

Reply:
> Please note that "Suggest" button doesn't care about such verification.

That is an explicit request to Suggest at least, you can avoid it. (If the tool is known in the gitconfig, the full command is filled in when typing the name so the default use case is supported).
You could argue that the check should be in `SuggestDiffToolCommand()` instead, but I believe that the `Suggest` button should be disabled in this case instead. Some more logic though.

The concept that GE knows about some tools improves usability some but makes the handling more complex...
(If path is quouted, the path is quoted again in command)